packages/ingest/granule.js

```
@@ -381,7 +363,21 @@ class Granule {
    */
   async verifyFile(file, bucket, key, options = {}) {
     const [type, value] = await this.retrieveSuppliedFileChecksumInformation(file);
-    if (!type || !value) return [null, null];
+    if (!type || !value) {
```

If both checksum and fileSize of the source file are provided, should we check both?

---

I used `fileSize` as a fallback to have at least a minimal check for a file without a provided checksum, as opposed to silently returning `null, null`. I'm not sure that a file with a matching checksum but a non-matching fileSize is a major enough edge case to check both. `fileSize` is generally more unreliable as a verifying factor so I'm not eager to rely on it, but I'm not really convinced which is the better move either way. Any specific thoughts on why or why not?

---

I don't have strong opinion on this.  But since fileSize mismatch is an error when checksum is not present, it should be an error as well if checksum is present.

---

I've made the change to always check fileSize as I could not think of a good reason not to.